internal/scripts/generate_docs.go

```diff
@@ -19,15 +19,26 @@ func run() error {
 	// Define the directory where the docs will be generated
 	dir := "../../docs/reference"
 
+	// Make sure the directory exists
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("error creating docs directory: %w", err)
+	}
+
 	// Clean the directory to make sure outdated files don't persist
-	err := os.RemoveAll(dir)
+	files, err := os.ReadDir(dir)
 	if err != nil {
-		return fmt.Errorf("could not remove docs directory: %w", err)
+		return fmt.Errorf("error listing doc files: %w", err)
 	}
-
-	// Create directory again
-	if err := os.MkdirAll(dir, 0755); err != nil {
-		return fmt.Errorf("error creating docs directory: %w", err)
+	for _, f := range files {
+		if !f.IsDir() && f.Name() == "README.md" {
+			// we want to keep the README.md file
+			continue
+		}
```

Comment:
Do we plan to add any other reference documentation in the future? If yes, we could move the autogenerated man pages to a subdirectory and only wipe that instead of keeping a "keep-list" of all other files.